L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc

```diff
@@ -0,0 +1,1199 @@
+                                            is_iso&& is_ss,
+                                            is_looseTkiso&& is_looseTkss,
+                                            is_iso&& is_ss);
+          // Experimental parameters, don't want to bother with hardcoding them in data format
+          std::map<std::string, float> params;
+          params["standaloneWP_showerShape"] = is_ss;
+          params["standaloneWP_isolation"] = is_iso;
+          params["trkMatchWP_showerShape"] = is_looseTkss;
+          params["trkMatchWP_isolation"] = is_looseTkiso;
+          params["TTiEta"] = getToweriEta_fromAbsoluteID(getTower_absoluteEtaID(p4calibrated.eta()));
+          params["TTiPhi"] = getToweriPhi_fromAbsoluteID(getTower_absolutePhiID(p4calibrated.phi()));
+          cluster.setExperimentalParams(params);
+          L1EGXtalClusters->push_back(cluster);
+
+          int standaloneWP = (int)(is_iso && is_ss);
+          int looseL1TkMatchWP = (int)(is_looseTkiso && is_looseTkss);
+          int photonWP = (int)(is_photon);
+          int quality =
+              (standaloneWP * std::pow(2, 0)) + (looseL1TkMatchWP * std::pow(2, 1)) + (photonWP * std::pow(2, 2));
+          L1EGammas->push_back(
+              0, l1t::EGamma(p4calibrated, p4calibrated.pt(), p4calibrated.eta(), p4calibrated.phi(), quality, 1));
+        }
+      }
+    }
+  }
+
+  for (int ii = 0; ii < n_links_GCTcard; ++ii) {  // 48 links
+    for (int ll = 0; ll < 3; ++ll) {              // 3 cards
+      // Fill the tower collection
+      for (int jj = 0; jj < n_towers_cardEta; ++jj) {  // 17 TTs
+        l1tp2::CaloTower l1CaloTower;
+        l1CaloTower.setEcalTowerEt(ECAL_tower_L2Card[ii][jj][ll]);
+        l1CaloTower.setHcalTowerEt(HCAL_tower_L2Card[ii][jj][ll]);
+        l1CaloTower.setTowerIEta(getToweriEta_fromAbsoluteID(iEta_tower_L2Card[ii][jj][ll]));
+        l1CaloTower.setTowerIPhi(getToweriPhi_fromAbsoluteID(iPhi_tower_L2Card[ii][jj][ll]));
+        l1CaloTower.setTowerEta(getTowerEta_fromAbsoluteID(iEta_tower_L2Card[ii][jj][ll]));
+        l1CaloTower.setTowerPhi(getTowerPhi_fromAbsoluteID(iPhi_tower_L2Card[ii][jj][ll]));
+        // Some towers have incorrect eta/phi because that wasn't initialized in certain cases, fix these
+        static float constexpr towerEtaUpperUnitialized = -80;
+        static float constexpr towerPhiUpperUnitialized = -90;
+        if (l1CaloTower.towerEta() < towerEtaUpperUnitialized && l1CaloTower.towerPhi() < towerPhiUpperUnitialized) {
+          l1CaloTower.setTowerEta(l1t::CaloTools::towerEta(l1CaloTower.towerIEta()));
+          l1CaloTower.setTowerPhi(l1t::CaloTools::towerPhi(l1CaloTower.towerIEta(), l1CaloTower.towerIPhi()));
+        }
+        l1CaloTower.setIsBarrel(true);
+
+        // Add L1EGs if they match in iEta / iPhi
+        // L1EGs are already pT ordered, we will take the ID info for the leading one, but pT as the sum
+        for (auto l1eg : *L1EGXtalClusters) {
```

Comment:
this is not the correct fix for the clang warning
it is telling you to use `const auto&` (reference, not copy)

Reply:
done